### ISSUE TITLE
Part Cooling Fan not shared between multiple extruders

### ISSUE THREAD

**Issue by Rhage** [bigtreetech/BIGTREETECH-SKR-PRO-V1.1#116]
I have a multi-extruder setup with a single nozzle

The Slicer (Cura 4.5) is set to turn the print cooling fan on starting at layer 2.
When layer 2 begins the third extruder is active and the part cooling fan correctly turns on.

However, as soon as the active extruder changes, to either extruder 0 or 1, the cooling fan turns off and only resumes when the third extruder (number 2) becomes the active extruder again.

This continues for the rest of the print. The print cooling fan turns on only for Extruder 2 and is off for extruder 0 and 1

Any help would be appreciated.

Screenshots of Slicer settings and my Marlin Configuration.h and Configuration_adv.h files are attached

![Slicer_Cooling](https://user-
[Configuration .txt](https://github.com/bigtreetech/BIGTREETECH-SKR-PRO-V1.1/files/4321897/Configuration.txt)
[Configuration_adv.txt](https://github.com/bigtreetech/BIGTREETECH-SKR-PRO-V1.1/files/4321898/Configuration_adv.txt)

images.githubusercontent.com/14357200/76479182-fc960f80-63d8-11ea-96ea-e66d4ddcdd1d.PNG)
![Slicer_Extruder_0](https://user-images.githubusercontent.com/14357200/76479184-fc960f80-63d8-11ea-88d6-29155bcf616b.PNG)
![Slicer_Extruder_1](https://user-images.githubusercontent.com/14357200/76479185-fd2ea600-63d8-11ea-95f6-4d43ddd78e4a.PNG)
![Slicer_Extruder_2](https://user-images.githubusercontent.com/14357200/76479186-fd2ea600-63d8-11ea-959b-88f74e21b569.PNG)
![Slicer_Machine_Settings](https://user-images.githubusercontent.com/14357200/76479187-fd2ea600-63d8-11ea-8c54-78890f67a6aa.PNG)

**Comment by oldman4U** [comment]
Hello Rhage.

Have you been able to solve the reported issue in the meantime?

Please let us know

**Comment by Rhage** [comment]
Yes, I was. But not in a way I'd probably advise duplicating.

It appears to be a bug/feature (depending on who you ask) of the Marlin 2.0 implementation of the m106 gcode

In my opinion it's a bug. The specification for that command states that when no fan parameter is provided with m106, which is how Cura sends this command with the setup I showed in my screenshots, it is supposed to default to Fan 0.

However, Marlin is defaulting the command with no fan parameter to the currently active tool.

I solved this by editing the Marlin source to always set the fan speed for m106 to all tool fans, which means my printer is no longer running the standard released version of Marlin 2.
